### PR TITLE
IGNITE-23437 Do not use ByteUtils#toBytes to persist timestamps in LowWatermark

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-23437

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java [hunk 1]
```diff
@@ -121,6 +122,15 @@ public static long hybridTimestampToLong(@Nullable HybridTimestamp timestamp) {
         return timestamp == null ? NULL_HYBRID_TIMESTAMP : timestamp.time;
     }
 
+    /**
+     * Restores a timestamp converted to bytes using {@link #toBytes()}.
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java:126]: Perhaps you wanted to indicate a different method?

File: modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java [hunk 2]
```diff
@@ -239,4 +249,11 @@ public HybridTimestamp roundUpToPhysicalTick() {
             return new HybridTimestamp(getPhysical() + 1, 0);
         }
     }
+
+    /**
+     * Returns a byte array representing this timestamp.
+     */
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java:255]:
```suggestion
    /** Returns a byte array representing this timestamp. */
```

File: modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java (hunk 1 above)
Comment (rpuch) [reply modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java:126]: No, I actually meant `HybridTimestamp#toBytes()`

File: modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java (hunk 2 above)
Comment (rpuch) [reply modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java:255]: On all methods of the class there are multi-line javadocs, so it doesn't make sense to use single-line javadoc here

File: modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java (hunk 1 above)
Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java:126]: Ok.

File: modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java (hunk 2 above)
Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java:255]: Ok.

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**